test/test_verifylib.py

```diff
@@ -584,6 +588,157 @@ def test_pass_verify_all_inspection_rules(self):
+
+    # dump expired layout
+    layout = copy.deepcopy(layout_template)
+    layout.expires = (datetime.today()
+       + relativedelta(months=-1)).isoformat() + "Z"
+    layout.sign(alice)
+    layout.dump(self.layout_expired_path)
+
+    # dump layout with failing step rule
+    layout = copy.deepcopy(layout_template)
+    layout.steps[0].material_matchrules.insert(0,
+        ["MATCH", "PRODUCT", "does-not-exist", "FROM", "write-code"])
+    layout.sign(alice)
+    layout.dump(self.layout_failing_step_rule_path)
+
+    # dump layout with failing inspection rule
+    layout = copy.deepcopy(layout_template)
+    layout.inspect[0].material_matchrules.insert(0,
+        ["MATCH", "PRODUCT", "does-not-exist", "FROM", "write-code"])
+    layout.sign(alice)
+    layout.dump(self.layout_failing_inspection_rule_path)
+
+  @classmethod
+  def tearDownClass(self):
+    """Change back to initial working dir and remove temp dir. """
+    os.chdir(self.working_dir)
+    shutil.rmtree(self.test_dir)
+
+  def test_verify_passing(self):
+    """Test pass verification of single-signed layout. """
+    in_toto_verify(self.layout_single_signed_path, [self.alice_path])
+
+  def test_verify_passing_double_signed_layout(self):
+    """Test pass verification of double-signed layout. """
+    in_toto_verify(self.layout_double_signed_path, [self.alice_path,
+        self.bob_path])
+
+  def test_verify_failing_layout_not_found(self):
+    """Test fail verification for layout not found. """
+    with self.assertRaises(IOError):
+      in_toto_verify("missing.layout", [self.alice_path])
+
+  def test_verify_failing_key_not_found(self):
+    """Test fail verification with layout key not found. """
+    with self.assertRaises(IOError):
+      in_toto_verify(self.layout_single_signed_path, ["missing-key.pub"])
+
+  def test_verify_failing_wrong_key(self):
+    """Test fail verification with wrong layout key. """
+    with self.assertRaises(SignatureVerificationError):
+      in_toto_verify(self.layout_single_signed_path, [self.bob_path])
+
+  def test_verify_failing_missing_key(self):
+    """Test fail verification with missing layout key. """
+    with self.assertRaises(SignatureVerificationError):
+      in_toto_verify(self.layout_double_signed_path, [self.bob_path])
+
+  def test_verify_failing_layout_expired(self):
+    """Test fail verification with expired layout. """
+    with self.assertRaises(LayoutExpiredError):
+      in_toto_verify(self.layout_expired_path, [self.alice_path, self.bob_path])
+
+  def test_verify_failing_link_metadata_files(self):
+    """Test fail verification with link metadata files not found. """
+    os.rename("package.link", "package.link.bak")
+    with self.assertRaises(IOError):
+      in_toto_verify(self.layout_single_signed_path, [self.alice_path])
+    os.rename("package.link.bak", "package.link")
+
+  def test_verify_failing_inspection_exits_non_zero(self):
+    """FIXME implement raise exception on inspection fail. """
+    pass
```

Comment:
Correct and correct. We discussed this with the VCS AppSec group, since they would have appreciated mentioned behavior. They planned to submit a ticket but never did.